client/authority-discovery/src/worker/tests.rs

```diff
@@ -474,97 +469,191 @@ fn dont_stop_polling_dht_event_stream_after_bogus_event() {
+
+		block_on(local_worker.refill_pending_lookups_queue()).unwrap();
+		local_worker.start_new_lookups();
+
+		drop(local_worker.handle_dht_value_found_event(values));
+
+		self.local_worker = Some(local_worker);
+
+		self.local_worker
+			.as_ref()
+			.map(|w| {
+				w.addr_cache
+					.get_addresses_by_authority_id(&self.remote_authority_public.clone().into())
+			})
+			.unwrap()
+	}
+}
+
 #[test]
 fn limit_number_of_addresses_added_to_cache_per_authority() {
-	let remote_key_store = KeyStore::new();
-	let remote_public =
-		block_on(remote_key_store.sr25519_generate_new(key_types::AUTHORITY_DISCOVERY, None))
-			.unwrap();
+	let mut tester = DhtValueFoundTester::new();
+	assert!(MAX_ADDRESSES_PER_AUTHORITY < 100);
+	let addresses = (1..100).map(|i| tester.multiaddr_with_peer_id(i)).collect();
+	let kv_pairs = block_on(build_dht_event(
+		addresses,
+		tester.remote_authority_public.clone().into(),
+		&tester.remote_key_store,
+		None,
+	));
 
-	let addresses = (0..100)
-		.map(|_| {
-			let peer_id = PeerId::random();
-			let address: Multiaddr = "/ip6/2001:db8:0:0:0:0:0:1/tcp/30333".parse().unwrap();
-			address.with(multiaddr::Protocol::P2p(peer_id.into()))
-		})
-		.collect();
+	let cached_remote_addresses = tester.process_value_found(false, kv_pairs);
+	assert_eq!(MAX_ADDRESSES_PER_AUTHORITY, cached_remote_addresses.unwrap().len());
+}
 
-	let dht_event = block_on(build_dht_event(addresses, remote_public.into(), &remote_key_store));
+#[test]
+fn strict_accept_address_with_peer_signature() {
+	let mut tester = DhtValueFoundTester::new();
+	let addr = tester.multiaddr_with_peer_id(1);
+	let kv_pairs = block_on(build_dht_event(
+		vec![addr.clone()],
+		tester.remote_authority_public.clone().into(),
+		&tester.remote_key_store,
+		Some(&tester.remote_node_key),
+	));
 
-	let (_dht_event_tx, dht_event_rx) = channel(1);
+	let cached_remote_addresses = tester.process_value_found(true, kv_pairs);
 
-	let (_to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
-		from_service,
-		Arc::new(TestApi { authorities: vec![remote_public.into()] }),
-		Arc::new(TestNetwork::default()),
-		Box::pin(dht_event_rx),
-		Role::Discover,
-		None,
-		Default::default(),
+	assert_eq!(
+		Some(&HashSet::from([addr])),
+		cached_remote_addresses,
+		"Expect worker to only cache `Multiaddr`s with `PeerId`s.",
 	);
+}
 
-	block_on(worker.refill_pending_lookups_queue()).unwrap();
-	worker.start_new_lookups();
+#[test]
+fn reject_address_with_rogue_peer_signature() {
+	let mut tester = DhtValueFoundTester::new();
+	let rogue_remote_node_key = libp2p::identity::Keypair::generate_ed25519();
+	let kv_pairs = block_on(build_dht_event(
+		vec![tester.multiaddr_with_peer_id(1)],
+		tester.remote_authority_public.clone().into(),
+		&tester.remote_key_store,
+		Some(&rogue_remote_node_key),
+	));
+
+	let cached_remote_addresses = tester.process_value_found(false, kv_pairs);
 
-	worker.handle_dht_value_found_event(vec![dht_event]).unwrap();
 	assert_eq!(
```

Comment:
```suggestion
	assert!(
```

Reply:
Before I do these, please compare the output in case the test fails, and I will commit whichever you like better.

## assert_eq!(None, x, "...")

```
---- worker::tests::strict_accept_address_with_peer_signature stdout ----
thread 'worker::tests::strict_accept_address_with_peer_signature' panicked at 'assertion failed: `(left == right)`
  left: `None`,
 right: `Some({"/ip6/2001:db8::1/tcp/30333/p2p/12D3KooWFz9VYFntQdAtKE7npxTHEeuZaiNNVVLquEohTDSVJfyN"})`: Expect worker to only cache `Multiaddr`s with `PeerId`s.', client/authority-discovery/src/worker/tests.rs:582:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## assert!(x.is_none(), "...")

```
---- worker::tests::strict_accept_address_with_peer_signature stdout ----
thread 'worker::tests::strict_accept_address_with_peer_signature' panicked at 'Expect worker to only cache `Multiaddr`s with `PeerId`s.', client/authority-discovery/src/worker/tests.rs:582:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I know the difference in the output and not sure how the first one should help me?

Reply:
You know which one slipped through the filter among those that should not without debugging it. But hey, you seem to have made a decision on this, so I will just commit the `.is_none()`